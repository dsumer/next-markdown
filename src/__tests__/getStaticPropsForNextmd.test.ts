import NextMarkdown from '..';

const nextMD = NextMarkdown({ pathToContent: 'src/__tests__/__filesystem__/' });

test('/about', async () => {
  expect(await nextMD.getStaticPropsForNextmd(['about'])).toMatchSnapshot();
});

test('/blog', async () => {
  expect(await nextMD.getStaticPropsForNextmd(['blog'])).toMatchSnapshot();
});

test('/path/to/e', async () => {
  expect(await nextMD.getStaticPropsForNextmd(['path', 'to', 'e'])).toMatchSnapshot();
});

test('an unknown path', async () => {
  await expect(nextMD.getStaticPropsForNextmd(['empty'])).rejects.toThrowError(
    Error('Could not find markdown file at path "empty"'),
  );
});

test('/docs/getting-started', async () => {
  expect(await nextMD.getStaticPropsForNextmd(['docs', 'getting-started'])).toMatchSnapshot();
});

test('/docs/api', async () => {
  expect(await nextMD.getStaticPropsForNextmd(['docs', 'api'])).toMatchSnapshot();
});

test('/docs/examples (no index.md)', async () => {
  await expect(nextMD.getStaticPropsForNextmd(['docs', 'examples'])).rejects.toThrowError(
    Error('Could not find markdown file at path "docs/examples"'),
  );
});
