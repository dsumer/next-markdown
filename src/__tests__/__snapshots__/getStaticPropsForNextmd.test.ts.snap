// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/about 1`] = `
Object {
  "props": Object {
    "frontMatter": Object {},
    "html": "",
    "markdown": "",
    "mdxSource": null,
    "nextmd": Array [
      "about",
    ],
    "subPaths": Array [],
    "tableOfContents": Array [],
  },
}
`;

exports[`/blog 1`] = `
Object {
  "props": Object {
    "frontMatter": Object {},
    "html": "",
    "markdown": "",
    "mdxSource": null,
    "nextmd": Array [
      "blog",
    ],
    "subPaths": Array [
      Object {
        "frontMatter": Object {},
        "markdown": "",
        "nextmd": Array [
          "blog",
          "post-1",
        ],
      },
      Object {
        "frontMatter": Object {},
        "markdown": "",
        "nextmd": Array [
          "blog",
          "post-2",
        ],
      },
      Object {
        "frontMatter": Object {},
        "markdown": "",
        "nextmd": Array [
          "blog",
          "post-3",
        ],
      },
    ],
    "tableOfContents": Array [],
  },
}
`;

exports[`/docs/api 1`] = `
Object {
  "props": Object {
    "frontMatter": Object {},
    "html": "",
    "markdown": "",
    "mdxSource": null,
    "nextmd": Array [
      "docs",
      "api",
    ],
    "subPaths": Array [
      Object {
        "frontMatter": Object {},
        "markdown": "",
        "nextmd": Array [
          "docs",
          "api",
          "get-static-paths",
        ],
      },
      Object {
        "frontMatter": Object {},
        "markdown": "",
        "nextmd": Array [
          "docs",
          "api",
          "get-static-props",
        ],
      },
    ],
    "tableOfContents": Array [],
  },
}
`;

exports[`/docs/getting-started 1`] = `
Object {
  "props": Object {
    "frontMatter": Object {},
    "html": "",
    "markdown": "",
    "mdxSource": null,
    "nextmd": Array [
      "docs",
      "getting-started",
    ],
    "subPaths": Array [
      Object {
        "frontMatter": Object {},
        "markdown": "",
        "nextmd": Array [
          "docs",
          "getting-started",
          "install",
        ],
      },
      Object {
        "frontMatter": Object {},
        "markdown": "",
        "nextmd": Array [
          "docs",
          "getting-started",
          "setup",
        ],
      },
    ],
    "tableOfContents": Array [],
  },
}
`;

exports[`/path/to/e 1`] = `
Object {
  "props": Object {
    "frontMatter": Object {},
    "html": null,
    "markdown": "",
    "mdxSource": "mocked",
    "nextmd": Array [
      "path",
      "to",
      "e",
    ],
    "subPaths": Array [],
    "tableOfContents": Array [],
  },
}
`;
